src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignContigsAndCallBreakpointsSpark.java

```diff
@@ -0,0 +1,121 @@
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.bwa.BwaSparkEngine;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import scala.Tuple2;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@CommandLineProgramProperties(summary="Align assembled contigs to the reference and call breakpoints from them.",
+        oneLineSummary="Align contigs and call breakpoints",
+        programGroup = SparkProgramGroup.class)
+public class AlignContigsAndCallBreakpointsSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Number of assemblies to process per task. We need to make sure that the work is not
+     * over-partitioned since each worker needs to localize the BWA reference index and read it into
+     * memory once per partition.
+     */
+    public static final int NUM_ASSEMBLIES_PER_PARTITION = 200;
+
+    @Argument(doc = "file for assembled breakpoint output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String output;
+
+    @Argument(doc = "Input file of assembled contigs", shortName = "inputFile",
+            fullName = "inputFile", optional = false)
+    private String input;
+
+    private static final Logger log = LogManager.getLogger(AlignContigsAndCallBreakpointsSpark.class);
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<String> inputAssemblies = ctx.textFile(input).cache();
+
+        final long numInputPartitions = inputAssemblies.count();
+
+        final JavaPairRDD<String, String> contigCollectionByBreakpointId =
+                inputAssemblies
+                        .flatMapToPair(AlignContigsAndCallBreakpointsSpark::splitAssemblyLine)
+                        .coalesce((int) Math.ceil((double) numInputPartitions / (double) NUM_ASSEMBLIES_PER_PARTITION));
+
+        final JavaPairRDD<String, RunSGAViaProcessBuilderOnSpark.ContigsCollection> breakpointIdsToContigsCollection =
+                contigCollectionByBreakpointId.mapValues(AlignContigsAndCallBreakpointsSpark::parsePackedFasta);
+
+        final JavaPairRDD<String, ContigAligner.AssembledBreakpoint> assembledBreakpoints = breakpointIdsToContigsCollection.mapPartitionsToPair(iter -> () -> {
+            try {
+                final String ref = referenceArguments.getReferenceFileName();
+                final File localRef = BucketUtils.isHadoopUrl(ref) ? BwaSparkEngine.localizeReferenceAndBwaIndexFiles(ref) : new File(ref);
```

Comment:
Ouch.  Spark doesn't make this easy, but I think there's considerable opportunity for improvement here:  Right now we're localizingReferenceAndBwsIndexFiles once per partition, if I'm reading this correctly.  Once per core would be better, once per executor would be better than that, and once per node would be better yet.  Maybe we can brainstorm?